crates/core/plugin_sm/src/plugin_manager.rs

```diff
@@ -96,7 +96,10 @@ impl ExternalPlugins {
             default_plugin_type: default_plugin_type.clone(),
             sudo,
         };
-        let () = plugins.load()?;
+        if let Err(e) = plugins.load() {
+            warn!("{:?}: {:?}", e.kind(), &plugins.plugin_dir);
+            return Ok(plugins);
```

Comment:
I think this one should stay as an error. This is true that the function fails to open the plugins because there is nothing.

Rather, we should change the logic of the line 204, not to return the error, and just to log `warn`.

```
        let plugins = Arc::new(Mutex::new(ExternalPlugins::open(
            self.config.sm_home.join("sm-plugins"),
            get_default_plugin(&self.config.config_location)?,
            Some("sudo".into()),
        )?));
```

Reply:
Not sure I follow, let's discuss this tomorrow

Reply:
I don't really consider the plugins directory being empty as an `Error`. So, I'm fine with a log. But if the plugins directory itself, or the plugins in that directory aren't readable, that could be considered an error.